packages/tasklist-client-node-js/src/lib/Types.ts

```diff
@@ -7,6 +7,12 @@ class TTaskState {
     CANCELED = literal`CANCELED`
 }
 
+export enum TaskStateREST {
+    COMPLETED = 'COMPLETED',
+    CREATED = 'CREATED',
+    CANCELED = 'CANCELED',
+}
+
```

Comment:
👋 🔴 This is the most controversial change in this PR.

## Why did I do this

This was the main issue with the failing specs. The `state` property sent in a task query was a different shape between the GraphQL endpoint and the REST endpoint.

The GraphQL endpoint expects an _object_, with a `value` property on it. The REST endpoint expects a string. When we sent a request to the REST endpoint with a literal object, it would respond with a Bad Request.

## What did I do

I updated the `TaskQuery.state` property below to allow either a LiteralObject or this new enum. This satisfies type-checking of the modified REST specs, which now pass in a `TaskStateREST` string instead of a `TaskState` LiteralObject.

## Is that the right thing to do?

I didn't want to create too much drift between the types, and I also wanted to push this up for feedback. I'm also curious what our plan is re: supporting both the REST and GraphQL clients. 

If we think we'll be ripping out the GraphQL client before v1, then I don't think there's any issue in what I've done, because it'll be undone by the time anyone uses it. 

If we plan to support both clients for the public, then I think I might need to create more drift between the types. This current change requires the user to know which type of TaskState/TaskStateREST to pass into the query, based on the client it's using. That is a foot-gun! Instead, maybe I should create a TaskQuery<T> where T is the type of the task state, and update each client to specify T.

Thoughts?

Reply:
A string literal union type might be the simplest if we only have to support REST, ie: `type TaskStateREST = 'COMPLETED' | 'CREATED' | 'CANCELED'`.

Maybe we remove GraphQL altogether before the release. The existing GraphQL package can stay out there and if anyone wants to maintain it, they can - but the supported SDK is REST only. That will reduce the technical complexity and maintenance.

Reply:
Addressed in latest commit!